### PR TITLE
Bump sklearn min version to 0.20

### LINKED ISSUE #383: Compat with scikit-learn dev
Handle removal of DeprecationDict (and the deprecation it was supporting).

### LINKED ISSUE #384: Re-enable warnings from sklearn
Disabled in https://github.com/dask/dask-ml/pull/382/files

### LINKED ISSUE #391: Require scikit-learn>=0.20
Any objections? I worry about confusion on things like `from dask_ml.compose import ColumnTransformer`, or some modules that have moved between 0.19 and 0.20.

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes #391 
Closes #383
Closes #384 

Subsumes #389

### CODE REVIEW

File: dask_ml/model_selection/_search.py [hunk 1]
```diff
@@ -65,8 +65,7 @@
 __all__ = ["GridSearchCV", "RandomizedSearchCV"]
 
 
-if SK_VERSION >= packaging.version.parse("0.19.1"):
-    from sklearn.utils.deprecation import DeprecationDict
+if SK_VERSION <= packaging.version.parse("0.21.dev0"):
```
Comment (TomAugspurger) [inline dask_ml/model_selection/_search.py]: FYI @jrbourbeau aaa0b31f4d894136bec475d2a96018bd808e2dea has your commit bringing in DeprecationDict, and I updated the logic here. I think the correct behavior is

1. Include the train-score keys and warn for 0.20.x
2. Do not include the train-score keys for >=0.21.dev0

Comment (jrbourbeau) [reply dask_ml/model_selection/_search.py]: LGTM, I agree with the logic here

**Comment by TomAugspurger** [comment]
Py36 and sklearn_dev passed earlier.

The py27 build stalled so I restarted it. https://circleci.com/gh/dask/dask-ml/2508?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by TomAugspurger** [comment]
That passed.

**Pull request merged**